packages/web/hooks/use-swap.ts

```diff
@@ -81,7 +85,50 @@ export function useSwap({
     useOtherCurrencies,
   });
 
-  const inAmountInput = useAmountInput(swapAssets.fromAsset);
+  const [gasAmount, setGasAmount] = useState<CoinPretty>();
+  const inAmountInput = useAmountInput({
+    currency: swapAssets.fromAsset,
+    gasAmount: gasAmount,
+  });
+
+  const {
+    data: maxBalanceQuote,
+    isLoading: isMaxBalanceQuoteLoading,
+    error: maxBalanceQuoteError,
+  } = useQueryRouterBestQuote(
+    {
+      tokenIn: swapAssets.fromAsset,
+      tokenOut: swapAssets.toAsset,
+      tokenInAmount: inAmountInput.balance?.toCoin().amount!,
+      forcePoolId: forceSwapInPoolId,
+      maxSlippage,
+    },
+    !!inAmountInput.balance && !inAmountInput.balance?.toDec().isZero()
+  );
+
+  const {
+    data: maxBalanceNetworkFee,
+    isLoading: isLoadingMaxBalanceNetworkFee,
+    error: maxBalanceNetworkFeeError,
+  } = useEstimateTxFees({
+    chainId: chainStore.osmosis.chainId,
+    messages: maxBalanceQuote?.messages,
+    enabled:
+      featureFlags.swapToolSimulateFee &&
+      !!inAmountInput.balance &&
+      !isMaxBalanceQuoteLoading,
+  });
+
+  const hasMaxBalanceError = useMemo(() => {
+    return !!maxBalanceNetworkFeeError || !!maxBalanceQuoteError;
+  }, [maxBalanceNetworkFeeError, maxBalanceQuoteError]);
+
+  useEffect(() => {
+    if (isNil(maxBalanceNetworkFee?.gasAmount)) return;
```

Comment:
Maybe it is better to move this effect to also be embedded in `useEstiamteTxFee` as optional callback, it will def reduce the clutter here.

Reply:
How about we create a separate hook, such as useSwapAmountInput, for this input flow? I'm reluctant to integrate the callback into useEstimateTxFee due to the added complexity of managing useEffect dependencies when they are nested

Reply:
Definitely a step in the right direction! Why not just allow  `useQueryRouterBestQuote` to chain `estimateTxFeesQueryFn` with an optional arg?